### PR TITLE
refactor(vouchers): Remove gear-pallet explicit dependency on vouchers

### PR DESCRIPTION
**Pull request by breathx**

### CODE REVIEW

**Review by gshep — APPROVED** [review]

File: pallets/gear-voucher/src/lib.rs [hunk 1]
```diff
@@ -124,8 +127,8 @@ pub mod pallet {
     // Gas pallet error.
     #[pallet::error]
     pub enum Error<T> {
-        FailureToCreateVoucher,
-        FailureToRedeemVoucher,
+        InsufficientBalance,
+        InvalidVoucher,
```
Comment (ark0f) [inline pallets/gear-voucher/src/lib.rs:131]: Too generic error. For example, we have `InvalidReservationId` which is overused in sys-calls

File: common/src/storage/complex/mailbox.rs [hunk 2]
```diff
@@ -99,6 +99,11 @@ pub trait MailboxError {
     fn element_not_found() -> Self;
 }
 
+impl MailboxError for () {
```
Comment (ark0f) [inline common/src/storage/complex/mailbox.rs:102]: Add comment this is for mock only

File: pallets/gear/src/tests.rs [hunk 3]
```diff
@@ -14295,7 +14295,7 @@ fn send_gasless_message_works() {
                     keep_alive: false,
                 }
             ),
-            Error::<Test>::FailureRedeemingVoucher
+            pallet_gear_bank::Error::<Test>::InsufficientBalance
```
Comment (ark0f) [inline pallets/gear/src/tests.rs:14298]:
```suggestion
            pallet_gear_voucher::Error::<Test>::InsufficientBalance
```

**Review by ark0f — APPROVED** [review]

File: pallets/gear/src/tests.rs (hunk 3 above)
Comment (breathx) [reply pallets/gear/src/tests.rs:14298]: Not possible yet, will be replaced in following PRs

File: common/src/storage/complex/mailbox.rs (hunk 2 above)
Comment (breathx) [reply common/src/storage/complex/mailbox.rs:102]: Pretty obvious cause its the same pattern for anything substrate-related

File: pallets/gear-voucher/src/lib.rs (hunk 1 above)
Comment (breathx) [reply pallets/gear-voucher/src/lib.rs:131]: Will be splitted into exact errors in following PRs as well

**Pull request merged**